proposals/NNNN-async-stream-continuation-hashable-conformance.md

```diff
@@ -0,0 +1,138 @@
+            continuation.yield(value)
+            self.continuations.insert(continuation)
+            continuation.onTermination = { _ in
+                DispatchQueue.main.async {
+                    self.continuations.remove(continuation)
+                }
+            }
+        }
+    }
+
+    private var continuations: Set<AsyncStream<Int>.Continuation> = []
+}
+```
+
+Without `Hashable` conformance, continuation needs to be associated with an artificial identifier,
+e.g. object identity by wrapping each continuation in a class:
+
+```swift
+@MainActor private class Sender {
+    var value: Int = 0 {
+        didSet {
+            for c in continuations {
+                c.value.yield(value)
+            }
+        }
+    }
+
+    var values: some AsyncSequence<Int, Never> {
+        AsyncStream<Int> { (continuation: AsyncStream<Int>.Continuation) -> Void in
+            continuation.yield(value)
+            let box = ContinuationBox(value: continuation)
+            self.continuations.insert(box)
+            continuation.onTermination = { _ in
+                DispatchQueue.main.async {
+                    self.continuations.remove(box)
+                }
+            }
+        }
+    }
+
+    private var continuations: Set<ContinuationBox> = []
+
+    private final class ContinuationBox: Hashable, Sendable {
+        let value: AsyncStream<Int>.Continuation
+
+        init(value: AsyncStream<Int>.Continuation) {
+            self.value = value
+        }
+
+        static func == (lhs: Sender.ContinuationBox, rhs: Sender.ContinuationBox) -> Bool {
+            lhs === rhs
+        }
+
+        func hash(into hasher: inout Hasher) {
+            hasher.combine(ObjectIdentifier(self))
+        }
+    }
+}
+```
+
+Note that capturing `continuation` or `box` in `onTermination` is safe, because `onTermination` is dropped after being called.
+And it is always called, even if `AsyncStream` is discarded without even being iterated.
+
+## Proposed solution
+
+Add `Hashable` conformance to `Async(Throwing)Stream.Continuation`.
+
+## Detailed design
+
+Every time when `build` closure of the `Async(Throwing)Stream.init()` is called,
+it receives a continuation distinct from all other continuations.
+All copies of the same continuation should compare equal.
+Yielding values or errors, finishing the stream or cancelling iteration should not affect equality.
+Assigning `onTermination` closures should not affect equality.
+
+## Source compatibility
+
+This is an additive change.
+
+Retroactive conformances are unlikely to exist, because current public API of the `Async(Throwing)Stream.Continuation` 
+does not provide anything that could be reasonably used to implement `Hashable` or `Equatable` conformances.
+
+## ABI compatibility
+
+This is an additive change. 
+
+## Implications on adoption
+
+Adopters will need a new version of the standard library.
+
+## Alternatives considered
+
+### Backdeployment
```

Comment:
Back-deployment is generally a platform-level concern anyway (i.e., at the platform owner's discretion) and so not typically considered as part of proposals.